tests/helpers/perf_run_helper.rb

```diff
@@ -461,25 +461,40 @@ def generate_scale_scenarios(gatling_scenario)
     file = File.read("#{scenarios_dir}/#{gatling_scenario}")
     json = JSON.parse(file)
 
-    # allow the base instances to be set via environment variable
+    # allow the settings to be set via environment variables
+    # TODO: refactor to streamline handling of multiple environment variables / settings
+    env_scale_class = ENV["PUPPET_SCALE_CLASS"]
     env_base_instances = ENV["PUPPET_GATLING_SCALE_BASE_INSTANCES"]
+    env_num_repetitions = ENV["PUPPET_GATLING_SCALE_NUM_REPETITIONS"]
     json_base_instances = json["nodes"][0]["num_instances"]
+    json_num_repetitions = json["nodes"][0]["num_repetitions"]
 
-    # TODO: refactor
     if !env_base_instances.nil?
       puts "Using environment specified base instances: #{env_base_instances}"
       @scale_base_instances = Integer(env_base_instances)
-
-      # update json
-      desc = "'role::by_size_small' role from perf control repo, #{@scale_base_instances} agents, 1 iteration"
-      json["run_description"] = desc
-      json["nodes"][0]["num_instances"] = @scale_base_instances
     else
       puts "Using JSON specified base instances: #{json_base_instances}"
       @scale_base_instances = json_base_instances
     end
 
+    if !env_num_repetitions.nil?
+      puts "Using environment specified number of repetitions: #{env_num_repetitions}"
+      @scale_num_repetitions = Integer(env_num_repetitions)
```

Comment:
Why is this an instance variable?

Reply:
Good question... I've updated `scale_setup` to make use of it :)